v2/pkg/operator/local_stored_collector.go

```diff
@@ -284,31 +290,33 @@ func (o LocalStorageCollector) prepareD2MCopyBatch(log clog.PluggableLoggerInter
 				return nil, err
 			}
 
-			if imgSpec.Transport != ociProtocol {
-				if imgSpec.IsImageByDigest() {
-					src = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent + ":" + imgSpec.Digest}, "/")
-					dest = strings.Join([]string{o.Opts.Destination, imgSpec.PathComponent + ":" + imgSpec.Digest}, "/")
-				} else {
-					src = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent}, "/") + ":" + imgSpec.Tag
-					dest = strings.Join([]string{o.Opts.Destination, imgSpec.PathComponent}, "/") + ":" + imgSpec.Tag
-				}
+			if img.Type == v1alpha2.TypeOperatorCatalog && len(img.TargetName) > 0 {
```

Comment:
Since we have a high number of conditionals here sometimes it is difficult to understand and it could lead to uncovered scenarios.

Nit: if possible to simplify to make it easier to understand.